include/mbedtls/build_info.h

```diff
@@ -80,115 +81,76 @@
 #include MBEDTLS_USER_CONFIG_FILE
 #endif
 
-/* Auto-enable MBEDTLS_MD_C if needed by a module that didn't require it
- * in a previous release, to ensure backwards compatibility.
- */
-#if defined(MBEDTLS_PKCS5_C)
-#define MBEDTLS_MD_C
+/* PSA crypto configuration */
+#if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
+#if defined(MBEDTLS_PSA_CRYPTO_CONFIG_FILE)
+#include MBEDTLS_PSA_CRYPTO_CONFIG_FILE
+#else
+#include "psa/crypto_config.h"
 #endif
-
-/* Auto-enable MBEDTLS_MD_LIGHT based on MBEDTLS_MD_C.
- * This allows checking for MD_LIGHT rather than MD_LIGHT || MD_C.
- */
-#if defined(MBEDTLS_MD_C)
-#define MBEDTLS_MD_LIGHT
+#if defined(MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE)
+#include MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE
 #endif
+#endif /* defined(MBEDTLS_PSA_CRYPTO_CONFIG) */
 
-/* Auto-enable MBEDTLS_MD_LIGHT if needed by a module that didn't require it
- * in a previous release, to ensure backwards compatibility.
+/* Adjust the PSA crypto configuration.
+ *
+ * Note that this must be done before "config_mbedtls_from_psa.h" since
+ * that header may take decisions based on symbols enabled here.
```

Comment:
This isn't quite right. `config_mbedtls_from_psa.h` is currently taking decisions based on symbols that are set here, but it shouldn't, really. And conversely what this header does should be enforced on the result of `config_psa_from_mbedtls.h`, because `config_psa_adjust.h` sets up invariants that some parts of the library assume.

Reply:
> And conversely what this header does should be enforced on the result of config_psa_from_mbedtls.h, because config_psa_adjust.h sets up invariants that some parts of the library assume.
Agreed and I believe I have addressed that in my last commit.

> config_mbedtls_from_psa.h is currently taking decisions based on symbols that are set here, but it shouldn't, really.
I am not sure about that. When MBEDTLS_PSA_CRYPTO_CONFIG is enabled and thus config_mbedtls_from_psa.h is used, it makes sense to me to adjust the PSA_WANT_ definitions before to include `config_mbedtls_from_psa.h`. That is still the case in the rebase I have pushed.